packages/calcite-components/src/components/tree/tree.tsx

```diff
@@ -382,8 +386,11 @@ export class Tree {
       nodeListToArray(this.el.querySelectorAll("calcite-tree-item")) as HTMLCalciteTreeItemElement[]
     ).filter((i) => i.selected);
 
-    this.calciteTreeSelect.emit();
+    if (event.detail.updateTarget) {
```

Comment:
I think `updateTarget` warrants its own local variable since it's referenced a few times.